pipeline/splicing/calcVarPriors.py

```diff
@@ -139,22 +168,314 @@ def getVarType(variant):
+        else:
+            if varGenPos <= donorStart and varGenPos >= donorEnd:
+                return True
+    return False
+
+
+def varInSpliceAcceptor(variant):
+    '''
+    Given a variant, determines if a variant is in reference transcript's splice acceptor region
+    splice acceptor region = 20 bases preceding exon and first 3 bases in exon
+    Returns True if variant in splice acceptor region
+    '''
+    varGenPos = int(variant["Pos"])
+    varStrand = getVarStrand(variant)
+    acceptorBounds = getRefSpliceAcceptorBoundaries(variant)
+    for exon in acceptorBounds.keys():
+        acceptorStart = acceptorBounds[exon]["acceptorStart"]
+        acceptorEnd = acceptorBounds[exon]["acceptorEnd"]
+        if varStrand == "+":
+            if varGenPos >= acceptorStart and varGenPos <= acceptorEnd:
+                return True
+        else:
+            if varGenPos <= acceptorStart and varGenPos >= acceptorEnd:
+                return True
+    return False
+
+
+def varInCiDomain(variant, boundaries):
+    '''
+    Given a variant, determines if variant is in a clinically important domain
+    Second argument determiens which boundaries (ENIGMA or PRIORS) are used for CI domains
+    Returns True if variant in CI domain
+    '''
+    varGenPos = int(variant["Pos"])
+    varGene = variant["Gene_Symbol"]
+    inExon = varInExon(variant)
+    if varGene == "BRCA1":
+        for domain in brca1CIDomains[boundaries].keys():
+            domainStart = brca1CIDomains[boundaries][domain]["domStart"]
+            domainEnd = brca1CIDomains[boundaries][domain]["domEnd"]
+            if varGenPos <= domainStart and varGenPos >= domainEnd and inExon == True:
+                return True
+    elif varGene == "BRCA2":
+        for domain in brca2CIDomains[boundaries].keys():
+            domainStart = brca2CIDomains[boundaries][domain]["domStart"]
+            domainEnd = brca2CIDomains[boundaries][domain]["domEnd"]
+            if varGenPos >= domainStart and varGenPos <= domainEnd and inExon == True:
+                return True
+    return False
+                
+
+def varInGreyZone(variant):
+    '''
+    Given a variant, determines if variant is before, in, or after the grey zone
+    Returns false if variant is before the grey zone
+    Returns "grey_zone_variant" if variant is in the grey zone
+    Return "after_grey_zone_variant" if variant is after the grey zone
+    '''
+    varGenPos = int(variant["Pos"])
+    varGene = variant["Gene_Symbol"]
+    inUTR = varInUTR(variant)
+    if varGene == "BRCA2":
+        greyZoneStart = greyZones[varGene]["greyZoneStart"]
+        greyZoneEnd = greyZones[varGene]["greyZoneEnd"]
+        if varGenPos >= greyZoneStart and varGenPos <= greyZoneEnd:
+            return "grey_zone_variant"
+        elif varGenPos > greyZoneEnd and inUTR == False:
+            return "after_grey_zone_variant"
+    return False
+                
+
+def getVarLocation(variant, boundaries):
+    '''
```

Comment:
again, the return statements herein will shortcut the function at any point. maybe this is intentional, but just want to double check

Reply:
it's intentional